From 653a440d4a1984f574e64963c8e8c55f05f25f87 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 13 Oct 2014 22:30:10 -0400
Subject: [misc] genwqe: fix uninitialized return value in genwqe_free_sync_sgl()

Message-id: <1413239416-7778-23-git-send-email-gduarte@redhat.com>
Patchwork-id: 97554
O-Subject: [RHEL7.1 PATCH BZ 1088491 22/28 v2] misc: genwqe: fix uninitialized return value in genwqe_free_sync_sgl()
Bugzilla: 1088491
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088491
Upstream Status: 63fa80cd582321693474f9cf9203329a2f8a8494

commit 63fa80cd582321693474f9cf9203329a2f8a8494
Author: Christian Engelmayer <cengelma@gmx.at>
Date:   Tue May 13 22:47:03 2014 +0200

    misc: genwqe: fix uninitialized return value in genwqe_free_sync_sgl()

    Function genwqe_free_sync_sgl() returns the value of variable 'rc'. 'rc' is
    only set in the error paths, thus initialize it by 0. Coverity CID 1204242.

    Signed-off-by: Christian Engelmayer <cengelma@gmx.at>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/genwqe/card_utils.c b/drivers/misc/genwqe/card_utils.c
index d049d27..5babf40 100644
--- a/drivers/misc/genwqe/card_utils.c
+++ b/drivers/misc/genwqe/card_utils.c
@@ -454,7 +454,7 @@ int genwqe_setup_sgl(struct genwqe_dev *cd, struct genwqe_sgl *sgl,
  */
 int genwqe_free_sync_sgl(struct genwqe_dev *cd, struct genwqe_sgl *sgl)
 {
- int rc;
+ int rc = 0;
  struct pci_dev *pci_dev = cd->pci_dev;
 
  if (sgl->fpage) {
-- 
1.7.1